src/compile.test.ts

```diff
@@ -140,12 +143,45 @@ it.each([
 );
 
 it.each([
-  ['an unrecognized binary file is imported', 'importBinary.js'],
-  ['a non-existent relative import is specified', 'relativeImportNotFound.js'],
-  ['unknown external imports are used', 'unknownExternalImport.js'],
-  ['an absolute import is specified', 'absoluteImport.js'],
-])('build fails when %s', (_, filename) =>
-  expect(compileFile(filename)).rejects.toThrowErrorMatchingSnapshot(),
+  [
+    'an unrecognized binary file is imported',
+    'importBinary.js',
+    `Failed to compile ${path.join(
+      process.cwd(),
+      'src',
+      '__test__',
+      'compile',
+      'randomData.js',
+    )}`,
+  ],
+  [
+    'a non-existent relative import is specified',
+    'relativeImportNotFound.js',
+    `Could not resolve './__doesNotExist.js' from ${path.join(
+      'src',
+      '__test__',
+      'compile',
+      'relativeImportNotFound.js',
+    )}`,
+  ],
+  [
+    'unknown external imports are used',
+    'unknownExternalImport.js',
+    `Compile failed.`,
+  ],
+  [
+    'an absolute import is specified',
+    'absoluteImport.js',
+    `/foo.js is imported by ${path.join(
+      process.cwd(),
+      'src',
+      '__test__',
+      'compile',
+      'absoluteImport.js',
+    )}, but absolute imports are disallowed`,
+  ],
+])('build fails when %s', (_, filename, errorMessage) =>
+  expect(compileFile(filename)).rejects.toThrowError(errorMessage),
```

Comment:
I moved from snapshots to inline the expectation, so it can be composed with `path.join`